pipit/trace.py

```diff
@@ -190,6 +190,78 @@ def _match_events(self):
 
             self.events = self.events.astype({"_matching_event": "Int32"})
 
+    def _match_messages(self):
+        """
+        Matches corresponding MpiSend/MpiRecv and MpiIsend/MpiIrecv instant events
+        """
+        if "_matching_event" not in self.events.columns:
+            self.events["_matching_event"] = None
+
+        if "_matching_timestamp" not in self.events.columns:
+            self.events["_matching_timestamp"] = np.nan
+
+        matching_events = list(self.events["_matching_event"])
+        matching_times = list(self.events["_matching_timestamp"])
+
+        mpi_events = self.events[
+            self.events["Name"].isin(["MpiSend", "MpiRecv", "MpiIsend", "MpiIrecv"])
+        ]
+
+        queue = [[] for _ in range(len(self.events["Process"].unique()))]
+
+        df_indices = list(mpi_events.index)
+        timestamps = list(mpi_events["Timestamp (ns)"])
+        names = list(mpi_events["Name"])
+        attrs = list(mpi_events["Attributes"])
+        processes = list(mpi_events["Process"])
+
+        # Iterate through all events
+        for i in range(len(mpi_events)):
```

Comment:
Change to iterate over send events and receive events separately, so that if a process with a receive event is iterated over, we can confirm that the send event has already been seen and added to the queue.